### PR TITLE
[Tool] Get a list of compile time for each C++ file

### PR DESCRIPTION
**Pull request by sduzh**
## Why I'm doing:
Now it takes a long time to compile BE, maybe we should try to reduce the compilation time, and the first step to solving the problem is to know where the compilation time is spent.

## What I'm doing:
Add a new build option `--save-compile-time` to get a list of compile time for each C++ file.
Another way to get the compile time for each file is to set the CMake property(`set_property(GLOBAL PROPERTY RULE_LAUNCH_COMPILE "${CMAKE_COMMAND} -E time")`), but the compile time from this method is mixed in with the compilation output, and additional work is required to get the summary result.

Example contents of `compile_times.txt` at the end of compilation:
```
9 /home/alex/code/starrocks/be/src/formats/parquet/group_reader.cpp
8 /home/alex/code/starrocks/be/src/formats/parquet/file_reader.cpp
13 /home/alex/code/starrocks/be/src/formats/parquet/page_reader.cpp
5 /home/alex/code/starrocks/be/src/formats/parquet/level_builder.cpp
1 /home/alex/code/starrocks/be/src/formats/orc/apache-orc/c++/src/BloomFilter.cc
4 /home/alex/code/starrocks/be/src/formats/parquet/column_chunk_writer.cpp
0 /home/alex/code/starrocks/be/src/formats/orc/apache-orc/c++/src/ByteRLE.cc
0 /home/alex/code/starrocks/be/src/formats/orc/apache-orc/c++/src/Common.cc
0 /home/alex/code/starrocks/be/src/formats/orc/apache-orc/c++/src/Exceptions.cc
2 /home/alex/code/starrocks/be/src/formats/orc/apache-orc/c++/src/sargs/SearchArgument.cc
1 /home/alex/code/starrocks/be/src/formats/orc/apache-orc/c++/src/ColumnPrinter.cc
1 /home/alex/code/starrocks/be/src/formats/orc/apache-orc/c++/src/MemoryPool.cc
1 /home/alex/code/starrocks/be/src/formats/orc/apache-orc/c++/src/Int128.cc
1 /home/alex/code/starrocks/be/src/formats/orc/apache-orc/c++/src/Compression.cc
0 /home/alex/code/starrocks/be/src/formats/orc/apache-orc/c++/src/Murmur3.cc
1 /home/alex/code/starrocks/be/src/formats/orc/apache-orc/c++/src/LzoDecompressor.cc
0 /home/alex/code/starrocks/be/src/formats/orc/apache-orc/c++/src/OrcFile.cc
1 /home/alex/code/starrocks/be/src/formats/orc/apache-orc/c++/src/ColumnReader.cc
0 /home/alex/code/starrocks/be/src/formats/orc/apache-orc/c++/src/RLEV2Util.cc
0 /home/alex/code/starrocks/be/src/formats/orc/apache-orc/c++/src/RLEv1.cc
1 /home/alex/code/starrocks/be/src/formats/orc/apache-orc/c++/src/RleDecoderV2.cc
1 /home/alex/code/starrocks/be/src/formats/orc/apache-orc/c++/src/RleEncoderV2.cc
1 /home/alex/code/starrocks/be/src/formats/orc/apache-orc/c++/src/RLE.cc
1 /home/alex/code/starrocks/be/src/formats/orc/apache-orc/c++/src/StripeStream.cc
1 /home/alex/code/starrocks/be/src/formats/orc/apache-orc/c++/src/Timezone.cc
0 /home/alex/code/starrocks/be/src/formats/orc/apache-orc/c++/src/Vector.cc
2 /home/alex/code/starrocks/be/src/formats/orc/apache-orc/c++/src/ColumnWriter.cc
1 /home/alex/code/starrocks/be/src/formats/orc/apache-orc/c++/src/Statistics.cc
1 /home/alex/code/starrocks/be/src/fs/fd_cache.cpp
4 /home/alex/code/starrocks/be/src/formats/orc/apache-orc/c++/src/wrap/orc-proto-wrapper.cc
1 /home/alex/code/starrocks/be/src/formats/orc/apache-orc/c++/src/TypeImpl.cc
2 /home/alex/code/starrocks/be/src/formats/orc/apache-orc/c++/src/Reader.cc
1 /home/alex/code/starrocks/be/src/formats/orc/apache-orc/c++/src/Writer.cc
1 /home/alex/code/starrocks/be/src/formats/orc/apache-orc/c++/src/bit_packing.cc
3 /home/alex/code/starrocks/be/src/fs/fs_posix.cpp
2 /home/alex/code/starrocks/be/src/fs/fs_util.cpp
2 /home/alex/code/starrocks/be/src/fs/fs_memory.cpp
2 /home/alex/code/starrocks/be/src/fs/hdfs/hdfs_fs_cache.cpp
3 /home/alex/code/starrocks/be/src/fs/fs_broker.cpp
3 /home/alex/code/starrocks/be/src/fs/fs_s3.cpp
4 /home/alex/code/starrocks/be/src/fs/fs_starlet.cpp
5 /home/alex/code/starrocks/be/src/fs/hdfs/fs_hdfs.cpp
2 /home/alex/code/starrocks/be/src/exec/sort_exec_exprs.cpp
8 /home/alex/code/starrocks/be/src/fs/fs.cpp
6 /home/alex/code/starrocks/be/src/exec/data_sink.cpp
7 /home/alex/code/starrocks/be/src/exec/empty_set_node.cpp
7 /home/alex/code/starrocks/be/src/exec/scan_node.cpp
8 /home/alex/code/starrocks/be/src/exec/exchange_node.cpp
8 /home/alex/code/starrocks/be/src/exec/select_node.cpp
7 /home/alex/code/starrocks/be/src/exec/short_circuit.cpp
6 /home/alex/code/starrocks/be/src/exec/tablet_info.cpp
2 /home/alex/code/starrocks/be/src/exec/mysql_scanner.cpp
6 /home/alex/code/starrocks/be/src/exec/tablet_sink_colocate_sender.cpp
13 /home/alex/code/starrocks/be/src/exec/exec_node.cpp
9 /home/alex/code/starrocks/be/src/exec/short_circuit_hybrid.cpp
5 /home/alex/code/starrocks/be/src/exec/tablet_sink_sender.cpp
7 /home/alex/code/starrocks/be/src/exec/tablet_sink_index_channel.cpp
10 /home/alex/code/starrocks/be/src/exec/olap_common.cpp
1 /home/alex/code/starrocks/be/src/exec/local_file_writer.cpp
2 /home/alex/code/starrocks/be/src/exec/es/es_scroll_query.cpp
3 /home/alex/code/starrocks/be/src/exec/es/es_query_builder.cpp
4 /home/alex/code/starrocks/be/src/exec/es/es_scan_reader.cpp
```

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [x] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

**Review by kevincai — DISMISSED** [review]

File: build-support/compile_time.sh [hunk 1]
```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/env bash
+# Licensed to the Apache Software Foundation (ASF) under one
```
Comment (imay) [inline build-support/compile_time.sh:2]: this is a new file, should use starrocks' license header.

File: build-support/compile_time.sh [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+# Modified from hyrise project
+
+output=`git rev-parse --show-toplevel`/compile_times.txt
```
Comment (imay) [inline build-support/compile_time.sh:21]: Better to add some comments to explain how this script works.

File: build.sh [hunk 3]
```diff
@@ -93,6 +93,9 @@ Usage: $0 <options>
      --with-clang-tidy  build Backend with clang-tidy(default without clang-tidy)
      --without-java-ext build Backend without java-extensions(default with java-extensions)
      -j                 build Backend parallel
+     --save-compile-time
```
Comment (imay) [inline build.sh:96]:
```suggestion
     --output-compile-time 
```

File: build-support/compile_time.sh (hunk 2 above)
Comment (sduzh) [reply build-support/compile_time.sh]: Fixed

File: build-support/compile_time.sh (hunk 1 above)
Comment (sduzh) [reply build-support/compile_time.sh]: Updated

File: build-support/compile_time.sh [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Modified from hyrise project
+
+# The root directory that contains .git, i.e, the root directory of the project
+output=$(git rev-parse --show-toplevel)/compile_times.txt
```
Comment (kevincai) [inline build-support/compile_time.sh:20]: one more thing, what if the script is running under the workspace that is not a git repository, just a snapshot copy of the source code.

Comment (sduzh) [reply build-support/compile_time.sh:20]: Sounds reasonable

**Review by kevincai — APPROVED** [review]

**Review by imay — APPROVED** [review]

**Pull request merged**